pkg/controllers/membercluster/v1beta1/membercluster_controller.go

```diff
@@ -169,15 +169,32 @@ func (r *Reconciler) handleDelete(ctx context.Context, mc *clusterv1beta1.Member
 		}
 		return runtime.Result{Requeue: true}, controller.NewUpdateIgnoreConflictError(r.updateMemberClusterStatus(ctx, mc))
 	}
+	if r.canForceDelete(mc, currentImc) {
+		klog.V(2).InfoS("Force delete the member cluster, by garbage collecting owned resources", "memberCluster", mcObjRef)
+		return runtime.Result{Requeue: true}, r.garbageCollect(ctx, mc)
+	}
 	klog.V(2).InfoS("Need to wait for the agent to leave", "memberCluster", mcObjRef, "joinedCondition", mcJoinedCondition)
 	// mark the imc as left to make sure the agent is leaving the fleet
 	if err := r.leave(ctx, mc, currentImc); err != nil {
 		klog.ErrorS(err, "Failed to mark the imc as leave", "memberCluster", mcObjRef)
 		return runtime.Result{}, err
 	}
 	// update the mc status to track the leaving status while we wait for all the agents to leave.
-	// once the imc is updated, the mc controller will reconcile again.
-	return runtime.Result{}, controller.NewUpdateIgnoreConflictError(r.updateMemberClusterStatus(ctx, mc))
+	// once the imc is updated, the mc controller will reconcile again ,or we reconcile to force delete
+	// the member cluster after force delete wait time.
+	return runtime.Result{RequeueAfter: r.ForceDeleteWaitTime}, controller.NewUpdateIgnoreConflictError(r.updateMemberClusterStatus(ctx, mc))
```

Comment:
why do we want to wait for so long?

Reply:
We didn't have to wait at all before, since we expect member agent to update IMC,

- If that occurs the logic remains the same
- if not, we wait until force delete wait time before requeue so that on the next reconcile we would have crossed the force delete wait time threshold and garbage collect